### DISCUSSION TITLE
SFS test

### DISCUSSION THREAD

**Discussion by 737simpilot** [CIDRAM/CIDRAM#509] [Q&A] [answered]
With the sheer amount of WAF rules I have in Cloudflare and how CIDRAM picks up the slack, it is extraordinarily rare to see if any spammer has tried to register on my website. So, I was wondering if the SFS module does in fact report in the log that they were blocked due to their IP or e-mail showing up at SFS, and if so is there anything in the API or documentation that allows me to test this functionality to make sure it works? Like perhaps a dummy test e-mail or something that would trigger CIDRAM's block and then log?

Thanks.

**Comment by Maikuolan** [comment, marked as answer]
> So, I was wondering if the SFS module does in fact report in the log that they were blocked due to their IP or e-mail showing up at SFS

If a request is blocked due to the SFS module, it'll be marked in the block reason for the block event, which is included in the logs. The module doesn't have its own dedicated field though. The module also doesn't check email addresses (currently, due to that the only time email addresses are likely to be accessible to CIDRAM in the request data, is when someone enters an email address in the field of a page form or something like that, and there's no reliable way for CIDRAM to know whether checking that information would be desirable in the first place, as there's no way for it to know the context of that information); it checks only IP addresses (the IP address of the request).

> and if so is there anything in the API or documentation that allows me to test this functionality to make sure it works?

Might need to determine some IP address already listed by SFS for that (e.g., by searching from the search page at the SFS website). For example, "1.2.3.4" (not sure about official testing IP addresses, but that's an arbitrary IP address I looked up just now to confirm whether there are any reports). Once you've determined an IP address to test against, from CIDRAM's own IP test page, if you enter that IP address, check the checkbox for modules, check the checkbox for testing as a sensitive request, uncheck everything else, and test; that should demonstrate whether the SFS module is performing lookups and blocking listed IP addresses correctly.

I *did*, however, find a bug in the SFS module while looking over code a few minutes ago, while preparing to reply here. The module *was performing* lookups correctly, but it *wasn't* correctly *processing* the results of those lookups. I've fixed that bug now, so, after updating to the latest version, we should notice better results from the module, hopefully. :-)

**Comment by 737simpilot** [comment]
Thanks. SFS has IP downloads and I had to test different IPs to get a proper test since many are already blocked by various other modules.
